opentracing-util/src/main/java/io/opentracing/util/ThreadLocalScope.java

```diff
@@ -48,7 +49,13 @@ public void close() {
             wrapped.finish();
         }
 
-        scopeManager.tlsScope.set(toRestore);
+        if (toRestore != null) {
+            scopeManager.tlsScope.set(toRestore);
+            scopeManager.listener.onActivate(toRestore.wrapped);
+        } else {
+            scopeManager.tlsScope.remove();
+            scopeManager.listener.onClose();
```

Comment:
Should be done earlier imo

Reply:
Are you sure? Listener code will then execute in the context of the previous span, not the one passed in as argument. Same applies to onClose.

Reply:
again, the "previous span" is not really relevant. This function closes the current scope, so it should invoke `listener.close(currentScope.span)`. Then it can do the housekeeping of the stack and re-activation of the previous span.

Reply:
Ok, maybe we misunderstood each other. Were you talking about onClose or onActivate?
`onClose` can be called before `scopeManager.tlsScope.remove();` and it makes sense. I'll update that.

Calling `onActivate` before `scopeManager.tlsScope.set(toRestore);` does not make sense - onActivate would run still with active previous span.

Reply:
On the second thought, what about renaming methods to `onActivated` and `onClosed`, and really call them both after change has happened? After all, this is listener to change, not interceptor. I would naturally expect, that state transition has already happened, when I'm receiving the call.

Reply:
sgtm

Reply:
Renamed